src/hooks/useFavorites.ts

```diff
@@ -0,0 +1,34 @@
+import { ImagePerBreed } from "../api/images/types";
+import { FAVORITES_CONST } from "../config/consts";
+import useStorageArray from "./useStorageArray";
```

Comment:
I saw it quite late, but since I had limited time to do the assignment so I decided to continue without this as I would need to create also some kind of user state, user management etc etc.

Local storage is one of the options. You can also choose session storage or cookie storage as options.

The pros of local storage are:
- that it's simple to implement and maintain, 
- there is no server load or authentication required
- and the favorites are persisted across sessions.

The cons are:
- local storage is vulnerable to XSS attacks,
- the favorites are stored on the device so if it's a shared computer then all of them will have the same favorites and if the user sits on another device he will not have the same favorites as well.
- Finally there is limited storage capacity on local storage

Reply:
Also, on the `at which level is the selection persisted` persistence is at the browser level and it's limited to a single device and browser instance. The favorites array will stay available until the user manually clears their browser cache.

If we wanted the array to be session specific we could easily achieve that by changing the storage type to useStorageArray to "sessionStorage".

The same way we could change it to "cookie" for cookie storage. 

If we wanted persistence across multiple devices bounded to a specific user then we need to use the favorites endpoint from the cats API and implement a way that the user selects a username in order to store his favorites.